### PR TITLE
wangle 2017.12.25.00 (new formula)

### PR DESCRIPTION
**Pull request by alebcay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/wangle.rb [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  sha256 "02abb153bc5be63a82d39fd4af879a09915ea2093412c749c86f878816ad0184"
+
+  depends_on "cmake" => :build
+  depends_on "folly"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "openssl"
+
+  def install
+    Dir.chdir("wangle")
```
Comment (DomT4) [inline Formula/wangle.rb]: `cd "wangle" do`

**Comment by alebcay** [comment]
Requested changes have been implemented.

File: Formula/wangle.rb [hunk 2]
```diff
@@ -0,0 +1,211 @@
+class Wangle < Formula
+  desc "Framework providing a set of common client/server abstractions"
+  homepage "https://github.com/facebook/wangle"
+  url "https://github.com/facebook/wangle/archive/v2017.12.25.00.tar.gz"
+  sha256 "02abb153bc5be63a82d39fd4af879a09915ea2093412c749c86f878816ad0184"
+
```
Comment (stephengroat) [inline Formula/wangle.rb]: maybe add a link to `head` with `head "https://github.com/facebook/wangle"`?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/wangle.rb [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  def install
+    cd "wangle" do
+      system "cmake", ".", *std_cmake_args
+      system "make", "install"
+    end
+  end
+
+  test do
+    (testpath/"EchoServer.cpp").write <<~EOS
```
Comment (ilovezfs) [inline Formula/wangle.rb]: let's fetch this and the other heredocs as resources

File: Formula/wangle.rb [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+          }
+        } catch (const std::exception& e) {
+          std::cout << exceptionStr(e) << std::endl;
+        }
+
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "-std=c++14",
+                    "-I#{Formula["gflags"].include}",
```
Comment (ilovezfs) [inline Formula/wangle.rb]: opt_include not include, opt_lib not lib

File: Formula/wangle.rb [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                    "-I#{Formula["openssl"].include}",
+                    "-I#{Formula["folly"].include}",
+                    "-I#{Formula["wangle"].include}",
+                    "-L#{Formula["gflags"].lib}",
+                    "-L#{Formula["glog"].lib}",
+                    "-L#{Formula["openssl"].lib}",
+                    "-L#{Formula["folly"].lib}",
+                    "-L#{Formula["wangle"].lib}",
+                    "EchoServer.cpp",
+                    "-lgflags", "-lglog", "-lcrypto", "-lssl", "-lfolly", "-lwangle",
```
Comment (ilovezfs) [inline Formula/wangle.rb]: probably want a `-L#{lib}` and `-I#{include}`

File: Formula/wangle.rb [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                    "-I#{Formula["folly"].include}",
+                    "-I#{Formula["wangle"].include}",
+                    "-L#{Formula["gflags"].lib}",
+                    "-L#{Formula["glog"].lib}",
+                    "-L#{Formula["openssl"].lib}",
+                    "-L#{Formula["folly"].lib}",
+                    "-L#{Formula["wangle"].lib}",
+                    "EchoClient.cpp",
+                    "-lgflags", "-lglog", "-lcrypto", "-lssl", "-lfolly", "-lwangle",
+                    "-o", "EchoClient"
```
Comment (ilovezfs) [inline Formula/wangle.rb]: we should actually execute the test program and assert that it does something

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@alebcay ping :)

**Pull request closed without merging**